dissect/target/plugins/os/unix/config.py

```diff
@@ -0,0 +1,264 @@
+from __future__ import annotations
+
+import io
+import re
+from abc import abstractmethod
+from configparser import ConfigParser, MissingSectionHeaderError
+from typing import Any, ItemsView, KeysView, Optional, TextIO, Union, BinaryIO, Iterator
+
+from dissect.target import Target
+from dissect.target.exceptions import ConfigurationParsingError
+from dissect.target.filesystem import Filesystem, FilesystemEntry, VirtualFilesystem
+from dissect.target.helpers import fsutil
+
+
+# TODO: Look if I can just create a parsing function and attach it to the
+# the parser below.
+class LinuxConfigurationParser:
+    def __init__(self, collapse: Optional[Union[bool, set]] = False) -> None:
+        self.collapse_all = collapse is True
+        self.collapse = collapse if isinstance(collapse, set) else {}
+        self.parsed_data = {}
+
+    @abstractmethod
+    def parse_file(self, fh: TextIO) -> None:
+        ...
+
+    def __getitem__(self, item: Any) -> Union[dict, str]:
+        return self.parsed_data[item]
+
+    def __contains__(self, item: str) -> bool:
+        return item in self.parsed_data
+
+    def read_file(self, fh: TextIO) -> None:
+        try:
+            self.parse_file(fh)
+        except Exception as e:
+            raise ConfigurationParsingError from e
+
+        if self.collapse_all or self.collapse:
+            self.parsed_data = self._collapse_dict(self.parsed_data)
+
+    def keys(self) -> KeysView:
+        return self.parsed_data.keys()
+
+    def items(self) -> ItemsView:
+        return self.parsed_data.items()
+
+    def _collapse_dict(self, dictionary: dict, collapse: bool = False) -> dict[str, dict]:
+        new_dictionary = {}
+
+        if isinstance(dictionary, list) and collapse:
+            return dictionary[-1]
+
+        if not hasattr(dictionary, "items"):
+            return dictionary
+
+        for key, value in dictionary.items():
+            value = self._collapse_dict(value, self.collapse_all or key in self.collapse)
+            new_dictionary.update({key: value})
+
+        return new_dictionary
+
+
+class Ini(LinuxConfigurationParser):
+    def __init__(self, collapse: Optional[Union[bool, set]] = True) -> None:
+        super().__init__(collapse)
+
+        self.parsed_data = ConfigParser(strict=False)
+        self.parsed_data.optionxform = str
+
+    def parse_file(self, fh: io.TextIO) -> None:
+        offset = fh.tell()
+        try:
+            self.parsed_data.read_file(fh)
+            return
+        except MissingSectionHeaderError:
+            pass
+
+        fh.seek(offset)
+        open_file = io.StringIO("[DEFAULT]\n" + fh.read())
+        self.parsed_data.read_file(open_file)
+
+
+class Unknown(LinuxConfigurationParser):
```

Comment:
Is this really unknown? It seems to me this is the standard Unix config style. I would recommend to name this Default instead.